rnd/autogpt_server/autogpt_server/blocks/ai.py

```diff
@@ -1,25 +1,59 @@
 import logging
 from enum import Enum
+from typing import NamedTuple
 
 import openai
+import anthropic
+from groq import Groq
 
 from autogpt_server.data.block import Block, BlockOutput, BlockSchema, BlockFieldSecret
 from autogpt_server.util import json
 
 logger = logging.getLogger(__name__)
 
+class ModelMetadata(NamedTuple):
+    provider: str
+    context_window: int
 
 class LlmModel(str, Enum):
-    openai_gpt4 = "gpt-4-turbo"
-
+    # OpenAI models
+    GPT4O = "gpt-4o"
+    GPT4_TURBO = "gpt-4-turbo"
+    GPT3_5_TURBO = "gpt-3.5-turbo"
+    # Anthropic models
+    CLAUDE_3_5_SONNET = "claude-3-5-sonnet-20240620"
+    CLAUDE_3_HAIKU = "claude-3-haiku-20240307"
+    # Groq models
+    LLAMA3_8B = "llama3-8b-8192"
+    LLAMA3_70B = "llama3-70b-8192"
+    MIXTRAL_8X7B = "mixtral-8x7b-32768"
+    GEMMA_7B = "gemma-7b-it"
+    GEMMA2_9B = "gemma2-9b-it"
+
+    @property
+    def metadata(self) -> ModelMetadata:
+        return MODEL_METADATA[self]
+
+MODEL_METADATA = {
+    LlmModel.GPT4O: ModelMetadata("openai", 128000),
+    LlmModel.GPT4_TURBO: ModelMetadata("openai", 128000),
+    LlmModel.GPT3_5_TURBO: ModelMetadata("openai", 16385),
+    LlmModel.CLAUDE_3_5_SONNET: ModelMetadata("anthropic", 200000),
+    LlmModel.CLAUDE_3_HAIKU: ModelMetadata("anthropic", 200000),
+    LlmModel.LLAMA3_8B: ModelMetadata("groq", 8192),
+    LlmModel.LLAMA3_70B: ModelMetadata("groq", 8192),
+    LlmModel.MIXTRAL_8X7B: ModelMetadata("groq", 32768),
+    LlmModel.GEMMA_7B: ModelMetadata("groq", 8192),
+    LlmModel.GEMMA2_9B: ModelMetadata("groq", 8192),
+}
 
 class LlmCallBlock(Block):
     class Input(BlockSchema):
         prompt: str
-        api_key: BlockFieldSecret = BlockFieldSecret(key="openai_api_key")
+        api_key: BlockFieldSecret
```

Comment:
@majdyz can you take a look at this? I may have broken server-side secrets

Reply:
This makes api_key becomes a mandatory input field. 
Can we change the key to something more generic? e.g: LLM_API_KEY and use just that ?